activerecord/lib/active_record/relation/query_attribute.rb

```diff
@@ -35,7 +35,7 @@ def with_cast_value(value)
       def nil?
         unless value_before_type_cast.is_a?(StatementCache::Substitute)
           value_before_type_cast.nil? ||
-            type.respond_to?(:subtype) && serializable? && value_for_database.nil?
+            (type.respond_to?(:subtype) || type.respond_to?(:normalizer)) && serializable? && value_for_database.nil?
```

Comment:
Annoys me a bit that we have to do so many `respond_to?` against our own code... But seems to work.